saleor/static/dashboard/js/components/async-form.js

```diff
@@ -6,20 +6,13 @@ export default $(document).on('submit', '.form-async', function (e) {
     url: $(that).attr('action'),
     method: 'post',
     data: $(that).serialize(),
-    complete: function (response) {
-      if (response.status === 400) {
-        $(that).parent().html(response.responseText);
-        initSelects();
-      } else {
-        $('.modal-close').click();
-      }
-    },
-    success: function (response) {
-      if (response.redirectUrl) {
-        window.location.href = response.redirectUrl;
-      } else {
-        location.reload();
-      }
+    success: function (data) {
+      let $message = $(data).find('div');
+      $message.find('a').remove();
+      $(that).find('.row').last().html($message);
+      $(that).find('.row').last().css('text-align', 'center');
+      $(that).find('svg').height('160px');
+      $(that).find('button').remove();
```

Comment:
Using `$(that)` is really not cool. Use `$(e.currentTarget)` beacuse JS `this` is not reliable.
Btw I think I refactored this code to ES6 some time ago.